contracts/DataFeeds/TokenPriceDataFeedFactory.sol

```diff
@@ -0,0 +1,48 @@
+pragma solidity >=0.4.24;
+
+import "./TokenPriceDataFeed.sol";
+
+contract TokenPriceDataFeedFactory {
+
+  event TokenPriceDataFeedCreated(address tokenPriceDataFeed, address token1, address token2);
+
+  IExchangeAdapter public exchangeAdapter;
+
+  // token1 => token2 => TokenPriceDataFeed
+  mapping(address => mapping(address => TokenPriceDataFeed)) public tokenPairToDataFeed;
+
+  constructor(IExchangeAdapter _exchangeAdapter) public {
+    exchangeAdapter = _exchangeAdapter;
+  }
+
+  function createTokenPriceDataFeed(
+    address token1,
+    address token2
+  ) public {
+    require(!tokenPriceDataFeedExists(token1, token2));
+
+    TokenPriceDataFeed tokenPriceDataFeed = new TokenPriceDataFeed();
+    tokenPriceDataFeed.initialize(token1, token2, exchangeAdapter);
+
+    tokenPairToDataFeed[token1][token2] = tokenPriceDataFeed;
+
+    emit TokenPriceDataFeedCreated(address(tokenPriceDataFeed), token1, token2);
+  }
+
+  function getTokenPriceDataFeed(address token1, address token2)
```

Comment:
not sure this matters, but having a public mapping and a public function to access the mapping feels redundant

Reply:
It's mostly a code readability thing .. also I was following the same pattern that DutchX contract uses since they're doing something similar with token pair mappings https://github.com/gnosis/dx-contracts/blob/master/contracts/DutchExchange.sol#L40